Assets/MixedRealityToolkit-SDK/Features/Diagnostics/MixedRealityDiagnosticsManager.cs

```diff
@@ -0,0 +1,243 @@
+﻿using Microsoft.MixedReality.Toolkit.Core.Definitions.Diagnostics;
+using Microsoft.MixedReality.Toolkit.Core.EventDatum.Boundary;
+using Microsoft.MixedReality.Toolkit.Core.Interfaces.Diagnostics;
+using Microsoft.MixedReality.Toolkit.Core.Managers;
+using Microsoft.MixedReality.Toolkit.Core.Utilities;
+using UnityEngine;
+using UnityEngine.EventSystems;
+
+namespace Assets.MixedRealityToolkit_SDK.Features.Diagnostics
+{
+    public class MixedRealityDiagnosticsManager : MixedRealityEventManager, IMixedRealityDiagnosticsManager
+    {
+        #region IMixedRealityManager
+        private DiagnosticsEventData eventData;
+        private GameObject diagnosticVisualization;
+
+        public override void Initialize()
+        {
+            base.Initialize();
+            InitializeInternal();
+        }
+
+        private void InitializeInternal()
+        {
+            eventData = new DiagnosticsEventData(EventSystem.current);
+
+            Visible = MixedRealityManager.Instance.ActiveProfile.DiagnosticsProfile.Visible;
+            ShowCpu = MixedRealityManager.Instance.ActiveProfile.DiagnosticsProfile.ShowCpu;
+            ShowFps = MixedRealityManager.Instance.ActiveProfile.DiagnosticsProfile.ShowFps;
+            ShowMemory = MixedRealityManager.Instance.ActiveProfile.DiagnosticsProfile.ShowMemory;
+
+            if (Visible)
+            {
+                GetDiagnosticVisualization();
+            }
+
+            RaiseDiagnosticsChanged();
+        }
+
+        /// <inheritdoc />
+        public override void Reset()
+        {
+            base.Reset();
+            InitializeInternal();
+        }
+
+        public override void Destroy()
+        {
+            base.Destroy();
+
+            if (Application.isPlaying)
+            {
+                if (diagnosticVisualization != null)
+                {
+                    if (Application.isEditor)
+                    {
+                        Object.DestroyImmediate(diagnosticVisualization);
+                    }
+                    else
+                    {
+                        Object.Destroy(diagnosticVisualization);
+                    }
+
+                    diagnosticVisualization = null;
+                }
+
+                visible = false;
+                showCpu = false;
+                showFps = false;
+                showMemory = false;
+
+                RaiseDiagnosticsChanged();
+            }
+        }
+
+        public GameObject GetDiagnosticVisualization()
+        {
+            if (diagnosticVisualization != null)
+            {
+                return diagnosticVisualization;
+            }
+
+            if (!Visible)
+            {
+                // Don't create a gameobject if it's not needed
+                return null;
+            }
+
+            diagnosticVisualization = GameObject.CreatePrimitive(PrimitiveType.Quad);
```

Comment:
Quad's aren't usually aligned properly, to face the camera (cause they're forward is perpendicular to their face). Are we going to add something for this later?

Reply:
Probably. This is my first dive into unity, so I just put something in that looked like it would work. Is there a primitive that makes more sense?

Reply:
Nah a quad is fine. I think it's okay for now until we do the UX pass. Just need to keep it in mind when we cross that bridge.